app/services/change_school_responsible_body_service.rb

```diff
@@ -31,14 +34,26 @@ def log_error(e)
     Sentry.capture_exception(e)
   end
 
-  def update_preorder_information!
-    school.preorder_information.refresh_status!
+  def remove_school_from_current_pool
+    initial_responsible_body.remove_school_from_virtual_cap_pools!(school)
+  end
+
+  def school_can_be_added?
+    new_responsible_body.can_school_be_added_to_virtual_cap_pools?(school)
+  end
+
+  def school_can_be_removed?
+    initial_responsible_body&.can_school_be_removed_from_virtual_cap_pools?(school)
   end
 
-  def update_school!
+  def set_school_new_responsible_body
```

Comment:
Does this method name need a bang `!`?

Reply:
Shouldn´t we add the bang only to the very internal method doing the dangerous thing? 
Otherwise we should add the ending bang to all the chain of methods calling that one?

Reply:
So for me, it's "more" a bang method than not...
1. It wraps update!
2. It changes the recipient and it returns the modified recipient.
3. It "feels" method that needs to be executed with caution or a warning.